res/controllers/MVave-SMK-25-II-scripts.js

```diff
@@ -0,0 +1,277 @@
+            });
+            // Deck Select
+            this.deckLeftButton = new components.Button({
+                group: "[Channel1]",
+                midi: [0x90, 0x2E], // << Channel Left Button
+                inToggle: function() {
+                    SMK25II.controller.activeDeck.setCurrentDeck("[Channel1]");
+                },
+            });
+            this.deckRightButton = new components.Button({
+                group: "[Channel2]",
+                midi: [0x90, 0x2F], // >> Channel Right button
+                inToggle: function() {
+                    SMK25II.controller.activeDeck.setCurrentDeck("[Channel2]");
+                },
+            });
+            // Drum Pads
+            this.samplers = [];
+            for (let i = 0; i < 16; i++) {
+                this.samplers[i] = new components.SamplerButton({
+                    number: i + 1,
+                    volumeByVelocity: true,
+                });
+            }
+        }
+    }
+
+    SMK25II.init = function() {
+        SMK25II.controller = new Controller();
+    };
+    SMK25II.shutdown = function() {
+        SMK25II.controller.shutdown();
+    };
+
+    const MMCHeader = [0xF0, 0x35, 0x59];
+
+    /**
+     *
+     * @param data the full sysex message data.
+     */
+    const decodeButton = function(data) {
+        // Hotcues
+        if (data[4] !== undefined && data[4] >= 0x00 && data[4] <= 0x07) {
+            SMK25II.controller.activeDeck.hotcues[data[4]].inToggle();
+            return;
+        }
+        // Other media buttons
+        switch (data[4]) {
+        case 0x5D: {
+            SMK25II.controller.activeDeck.cueButton.inToggle();
+            break;
+        }
+        case 0x5E: {
+            if (data[5] === 0x00) {
+                SMK25II.controller.activeDeck.playButton.inToggle();
+            }
+            break;
+        }
+        case 0x5B: {
+            SMK25II.controller.activeDeck.backButton.inToggle();
+            break;
+        }
+        case 0x5C: {
+            SMK25II.controller.activeDeck.forwardButton.inToggle();
+            break;
+        }
+        case 0x5F: {
+            SMK25II.controller.recordButton.inToggle();
+            break;
+        }
+        case 0x2E: {
+            SMK25II.controller.deckLeftButton.inToggle();
+            break;
+        }
+        case 0x2F: {
+            SMK25II.controller.deckRightButton.inToggle();
+            break;
+        }
+        case 0x4C: {
+            SMK25II.controller.activeDeck.loopButton.inToggle();
+            break;
+        }
+        default:
+            console.log(`unrecognized MMC command: ${data[4]}`);
+        }
+    };
+
+    /**
+     *
+     * @param data the full sysex message data.
+     */
+    const decodePots = function(data) {
+        switch (data[3]) {
+        case 0x60: {
+            const knob = SMK25II.controller.activeDeck.gainKnob;
+            knob.inSetParameter(knob.inValueScale(data[5]));
```

Comment:
also wouldn't recommend this. both functions are supposed to be private. Just dispatch to `input` and pass `undefined` to the non-applicable parameters.

Reply:
Done; thank you!